### PR TITLE
RFC: doc: mcumgr: smp: Add install_progress key to responses

### PR DESCRIPTION
**Pull request by de-nordic**
The commit documents new, optional, key to image status and image
upload responses that show state of post upload installation progress.
The field is optional and will only be returned by apps that require
post upload, installation, step.

2022-09-15 Update:
  Removed notes on "rc" return values. Previously the PR proposed returning "rc" code of value `MGMT_ERR_EBUSY` to indicate that there is still some operation in progress, but the MGMT codes should only be used to report  SMP protocol processing errors.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst [hunk 1]
```diff
@@ -304,6 +322,9 @@ to continue it in case when it gets broken.
 If library gets request with "off" equal zero it checks stored "sha" within its
 state and if it matches it will respond to update client application with
 offset that it should continue with.
+In case when application supports image post processing, and responses return
+"install_progress" value, after receiving final response with "off" eqal
+to image size, application can keep sending
```
Comment (nvlsianpu) [inline doc/services/device_mgmt/smp_groups/smp_group_1.rst:327]: `can keep sending` : [wording] subject is missing

**Review by nvlsianpu — COMMENTED** [review]
This protocol extension looks good to me.

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst (hunk 1 above)
Comment (de-nordic) [reply doc/services/device_mgmt/smp_groups/smp_group_1.rst:327]: Will fix.

**Review by nvlsianpu — COMMENTED** [review]
@de-nordic I'm awaiting for protocol implementation part :)

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst [hunk 2]
```diff
@@ -179,6 +180,23 @@ where:
...
     |                       | with application part; this is unused by Zephyr   |
     +-----------------------+---------------------------------------------------+
+    | "install_progress"    | Optional key returned to indicate that image will |
+    |                       | be post processed after upload or that post       |
+    |                       | processing is in progress, taking value of        |
+    |                       | percentage of completeness or negative percentage |
+    |                       | to indicate failure.                              |
+    |                       | When post processing is required this should be   |
+    |                       | with first successful response, taking value of 0 |
+    |                       | as indication that post processing will be        |
```
Comment (nordicjm) [inline doc/services/device_mgmt/smp_groups/smp_group_1.rst:190]: *will be performed
(or similar)

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst [hunk 3]
```diff
@@ -179,6 +180,23 @@ where:
...
+    |                       | although has not yet started;                     |
+    |                       | When upload completes, it will take value 1-99    |
+    |                       | indicating that post processing takes place       |
+    |                       | together with "rc" code equal ``10`` (busy);      |
+    |                       | 100 to indicates completion with "rc" equal ``0`` |
```
Comment (nordicjm) [inline doc/services/device_mgmt/smp_groups/smp_group_1.rst:195]: *indicate

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst [hunk 4]
```diff
@@ -344,6 +367,23 @@ where:
...
     |                       | unknown error                                     |
     +-----------------------+---------------------------------------------------+
+    | "install_progress"    | Optional key returned to indicate that image will |
+    |                       | be post processed after upload or that post       |
+    |                       | processing is in progress, taking value of        |
+    |                       | percentage of completeness or negative percentage |
+    |                       | to indicate failure.                              |
+    |                       | When post processing is required this should be   |
+    |                       | with first successful response, taking value of 0 |
+    |                       | as indication that post processing will be        |
```
Comment (nordicjm) [inline doc/services/device_mgmt/smp_groups/smp_group_1.rst:377]: Update as above

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst (hunk 4 above)
Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_1.rst:375]: *post processing will be peformed

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: doc/services/device_mgmt/smp_groups/smp_group_1.rst (hunk 3 above)
Comment (nordicjm) [reply doc/services/device_mgmt/smp_groups/smp_group_1.rst]: Needs fixing, not plural

**Pull request closed without merging**